test/user.test.js

```diff
@@ -1418,6 +1418,21 @@ describe('User', function() {
         }, done);
       });
 
+      it('sets verificationToken to null after confirmation', function(done) {
+        testConfirm(function(result, done) {
+          User.confirm(result.uid, result.token, false, function(err) {
+            if (err) return done(err);
+
+            // Verify by loading user data stored in the datasource
+            User.findById(result.uid, function(err, user) {
+              if (err) return done(err);
+              expect(user).to.have.property('verificationToken', null);
+              done();
+            });
+          });
+        }, done);
+      });
+
       it('Should report 302 when redirect url is set', function(done) {
```

Comment:
You are calling done twice, see L1440. More importantly, this test passes even when common/models/user.json is not fixed.

_Never trust a test you haven't seen fail._

When I added console logs around `done()` calls, only the second done on L1440 was called. The event handler for `confirmed` even is never triggered. I think that's because you are adding the listener long time after `confirm()` method was invoked.

I am proposing to simplify this test and call `User.confirm` directly - the behaviour you are fixing is independent of the way how the method is invoked (from code or via REST API).

As I was playing with the test code to make it work correctly, I figured out I can as well commit my changes - see fc90538. This fixup commit should be rebased away before merging the PR.